src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -820,7 +820,13 @@ public override bool HandleCall (MethodBody callingMethodBody, MethodReference c
 								foreach (var stringParam in methodParams[1].UniqueValues ()) {
 									if (stringParam is KnownStringValue stringValue) {
 										foreach (var method in systemTypeValue.TypeRepresented.GetMethodsOnTypeHierarchy (m => m.Name == stringValue.Contents, bindingFlags)) {
-											ValidateGenericMethodInstantiation (ref reflectionContext, method, methodParams[2], calledMethod);
+											if (!ValidateGenericMethodInstantiation (ref reflectionContext, method, methodParams[2], calledMethod)) {
+												reflectionContext.RecordUnrecognizedPattern (
+												2060, string.Format (Resources.Strings.IL2060,
+													DiagnosticUtilities.GetMethodSignatureDisplayName (calledMethod)));
+											} else {
+												reflectionContext.RecordHandledPattern ();
+											}
```

Comment:
Why not do this inside the ValidateGenericMethodInstantiation - the two callsites look exactly the same.

Reply:
Initially I was going to reuse `ValidateGenericMethodInstantiation` with the `MakeGenericType` case, but I've since realized that doesn't quite work right. I'll refactor this into there.